ghost_code.py

```diff
@@ -359,7 +386,53 @@ class TemporaryFunctionSignature:
     postcondition: source.ExprT[source.ProgVarName | nip.GuardVarName]
 
 
-def sprinkle_ghost_code(filename: str, func: nip.Function, unsafe_ctx: Mapping[str, syntax.Function]) -> Function:
+def sprinkle_handlerloop(fn: Function) -> Function:
```

Comment:
this is definitely no good but I cleaned it up (got rid of it) on my branch (which is based on this branch).